src/main/java/org/terasology/math/geom/Polygon.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2014 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.math.geom;
+
+import java.util.List;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+/**
+ * @author Martin Steiger
+ */
+public final class Polygon implements Shape {
+
+    private final List<ImmutableVector2f> vertices;
+
+    /**
+     * @param vertices a list of vertices (copied for internal storage)
+     */
+    public Polygon(List<? extends BaseVector2f> vertices) {
+        Builder<ImmutableVector2f> bldr = ImmutableList.builder();
+        for (BaseVector2f v : vertices) {
+            if (v instanceof ImmutableVector2f) {
+                bldr.add((ImmutableVector2f) v);
+            } else {
+                bldr.add(new ImmutableVector2f(v));
+            }
+        }
+
+        this.vertices = bldr.build();
+    }
+
+    /**
+     * @return the area of the polygon
+     */
+    public double area() {
+        return Math.abs(signedDoubleArea() * 0.5);
+    }
+
+    /**
+     * @return the winding of the polygon
+     */
+    public Winding winding() {
+        double signedDoubleArea = signedDoubleArea();
+        if (signedDoubleArea < 0) {
+            return Winding.CLOCKWISE;
+        }
+        if (signedDoubleArea > 0) {
+            return Winding.COUNTERCLOCKWISE;
+        }
+        return Winding.NONE;
+    }
+
+    private double signedDoubleArea() {
```

Comment:
Here, double means that the return value is twice the polygon area. I changed it to avoid misunderstandings.